### PR TITLE
logictest: add a kvtrace test type to logic tests.

### PR DESCRIPTION
**Pull request by jordanlewis**
It's common to want to test "kv trace" output in execbuilder tests, to
make sure that a particular SQL statement emits the expected set of key
reads, writes, deletes, and so on. It's very annoying to re-type out the
setup for kv tracing every time (it's multiple boiler plate-y lines), so
this commit introduces a shortcut for this common operation.

It looks like this:

```
query T kvtrace
SELECT * FROM t
----
<expected kv trace output>
```

Release note: None

### CODE REVIEW

**Comment by rohany** [comment]
Oh this is pretty slick -- is it possible to do something like
```
query T kvtrace(CPut, Get, InitPut)
...
```
To only see the requested operations?

Also, would it be too special cased to have the command accept a table id or an index id to filter on?

File: pkg/sql/logictest/logic.go [hunk 1]
```diff
@@ -1658,6 +1664,32 @@ func (t *logicTest) processSubtest(
...
+					_, err = t.db.Exec(query.sql)
+					if err != nil {
+						t.Error(err)
+					}
+					_, err = t.db.Exec("SET TRACING=off")
+					if err != nil {
+						return err
+					}
+					query.colTypes = "T"
+					query.sql = `SELECT message FROM [SHOW KV TRACE FOR SESSION] WITH ORDINALITY
```
Comment (rohany) [inline pkg/sql/logictest/logic.go]: Why do you need `WITH ORDINALITY` here?

**Comment by rohany** [comment]
Might also be worth it to make an issue tracking transferring all our existing tests that do the boilerplate to use this

File: pkg/sql/logictest/logic.go (hunk 1 above)
Comment (jordanlewis) [reply pkg/sql/logictest/logic.go]: I don't know, I just cargo culted it from somewhere else :) I'll get rid of it.

**Comment by jordanlewis** [comment]
As far as enhancing the test type to include expected op types, I think it's a little too annoying at the moment given the way the logic test parser works. I could do it later maybe but for now I don't really want to bother, since it'll probably work for most use cases at the moment just fine.

**Comment by jordanlewis** [comment]
I agree it's a good idea to transfer the existing tests to use this but yeah I definitely don't want to do this right now.

**Comment by rohany** [comment]
Sweet, both of those seem reasonable to me. code LGTM

**Pull request merged**